src/waterfall/waterfallplot.cpp

```diff
@@ -273,7 +259,10 @@ void WaterfallPlot::draw(const QVector<double>& points, float confidence, float
             _image.setPixelColor(_currentDrawIndex, i + virtualFloor, valueToRGB(points[factor * i]));
         }
     }
-    _currentDrawIndex++; // This can get to be an issue at very fast update rates from ping
+
+    for (int i = virtualHeight; i < _image.height(); i++) {
```

Comment:
I'm not sure how fast is this but perhaps this is something that *could* be improved by QConcurrent?